cmd/status/printers/table_printer.go

```diff
@@ -0,0 +1,385 @@
+						_, err = fmt.Fprintf(w, ",")
+						if err != nil {
+							return realLength, err
+						}
+						realLength += 1
+					}
+				}
+				return realLength, nil
+			},
+		},
+		{
+			// Column shows the age of the resource. This is computed based on the value of
+			// the creationTimestamp field.
+			name:   "age",
+			header: "AGE",
+			width:  6,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				u := resource.Resource
+				if u == nil {
+					return fmt.Fprint(w, "-")
+				}
+
+				timestamp, found, err := unstructured.NestedString(u.Object, "metadata", "creationTimestamp")
+				if !found || err != nil || timestamp == "" {
+					return fmt.Fprint(w, "-")
+				}
+				parsedTime, err := time.Parse(time.RFC3339, timestamp)
+				if err != nil {
+					return fmt.Fprint(w, "-")
+				}
+				age := time.Since(parsedTime)
+				switch {
+				case age.Seconds() <= 90:
+					return fmt.Fprintf(w, "%ds", integer.RoundToInt32(age.Round(time.Second).Seconds()))
+				case age.Minutes() <= 90:
+					return fmt.Fprintf(w, "%dm", integer.RoundToInt32(age.Round(time.Minute).Minutes()))
+				default:
+					return fmt.Fprintf(w, "%dh", integer.RoundToInt32(age.Round(time.Hour).Hours()))
+				}
+			},
+		},
+		{
+			// Column shows more detailed information regarding the status of the resource. If there
+			// was an error encountered while fetching the resource or computing status, the error
+			// message is shown in this column.
+			name:   "message",
+			header: "MESSAGE",
+			width:  40,
+			printContent: func(w io.Writer, availableWidth int, resource *event.ResourceStatus) (int, error) {
+				var message string
+				if resource.Error != nil {
+					message = resource.Error.Error()
+				} else {
+					message = resource.Message
+				}
+				if len(message) > availableWidth {
+					message = message[:availableWidth]
+				}
+				return fmt.Fprint(w, message)
+			},
+		},
+	}
+)
+
+// tablePrinter is an implementation of the Printer interface that outputs status information
+// about resources in a table format with in-place updates.
+type tablePrinter struct {
+	collector *collector.ResourceStatusCollector
+	w         io.Writer
+}
+
+// NewTablePrinter returns a new instance of the tablePrinter. The passed in collector is the
+// source of data to be printed, and the writer is where the printer will send the output.
+func NewTablePrinter(collector *collector.ResourceStatusCollector, w io.Writer) *tablePrinter {
+	return &tablePrinter{
+		collector: collector,
+		w:         w,
+	}
+}
+
+// PrintUntil prints the table of resources with their statuses until the
+// provided stop channel is closed.
+func (t *tablePrinter) PrintUntil(stop <-chan struct{}) <-chan struct{} {
+	completed := make(chan struct{})
+
+	linesPrinted := t.printTable(t.collector.LatestObservation(), 0)
+
+	go func() {
+		defer close(completed)
+		ticker := time.NewTicker(updateInterval)
+		for {
+			select {
```

Comment:
Maybe also add deadline timer here?  Or is that what's sitting behind 'stop'?

Reply:
Yeah, the stop channel is used to break out of the loop here. That allows the caller to stop the printer for whatever reason, not only based on a timeout.